presto-hive/src/main/java/io/prestosql/plugin/hive/s3/PrestoS3FileSystem.java

```diff
@@ -674,6 +677,20 @@ private AmazonS3 createAmazonS3Client(Configuration hadoopConfig, ClientConfigur
             clientConfig.withSignerOverride(signerType);
         }
 
+        String signerClass = hadoopConfig.get(S3_SIGNER_CLASS);
```

Comment:
this might be called multiple times. I think we would rather have signer class registered once. That's why `io.prestosql.plugin.hive.s3.PrestoS3ConfigurationInitializer` seems like a better place

Reply:
I've made it idempotent.
com.acme.CustomSigner will become ComAcmeCustomSignerType.

Reply:
> com.acme.CustomSigner will become ComAcmeCustomSignerType.

`SignerFactory.registerSigner(customSignerType, klass);` is still called multiple times (it does simple put in map at the moment, but could as well check that signer is not already registered).

I think the previous approach was easier

Reply:
This `String customSignerType = "signer" + randomUUID().toString().toLowerCase(ENGLISH).replace("-", "");`

could be replaced by `String customSignerType = "PrestoS3CustomSigner"`

Reply:
Done